src/python/pants/backend/jvm/tasks/ivy_updates.py

```diff
@@ -0,0 +1,91 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+import re
+
+from pants.backend.jvm.ivy_utils import IvyUtils
+from pants.backend.jvm.subsystems.shader import Shader
+from pants.backend.jvm.targets.jar_dependency import JarDependency
+from pants.backend.jvm.tasks.nailgun_task import NailgunTask
+from pants.base.workunit import WorkUnitLabel
+from pants.ivy.ivy_subsystem import IvySubsystem
+from pants.util.memo import memoized_property
+
+
+class IvyUpdates(NailgunTask):
+
+  _IVY_DEPENDENCY_UPDATE_MAIN = 'org.pantsbuild.tools.ivy.DependencyUpdateChecker'
+
+  @classmethod
+  def subsystem_dependencies(cls):
+    return super(IvyUpdates, cls).subsystem_dependencies() + (IvySubsystem.scoped(cls),)
+
+  @classmethod
+  def register_options(cls, register):
+    super(IvyUpdates, cls).register_options(register)
+    register('--confs', type=list, default=['default'],
+             help='Pass a configuration to ivy in addition to the default ones.')
+    register('--exclude-coordinates', type=list, default=[], fingerprint=True,
+             help='Adds patterns for excluded jar coordinates.')
+
+    cls.register_jvm_tool(register,
+                          'dependency-update-checker',
+                          classpath=[
+                            # JarDependency(org='org.pantsbuild',
+                            #               name='ivy-depdendency-update-checker',
+                            #               rev='0.0.1'),
+                            # First run ./pants binary src/java/org/pantsbuild/tools/ivy:ivy-dependency-update-checker
+                            JarDependency(org = 'org.pantsbuild',
+                                          name = 'ivy-depdendency-update-checker',
+                                          rev = 'none',
+                                          mutable = True,
+                                url = 'file:///Users/cheister/Development/pants/dist/ivy-dependency-update-checker.jar'),
+                          ],
+                          main=cls._IVY_DEPENDENCY_UPDATE_MAIN,
+                          custom_rules=[
+                            Shader.exclude_package('org.apache.ivy', recursive=True)
+                          ]
+                          )
+
+  @memoized_property
+  def _exclude_coordinates(self):
+    return "(" + ")|(".join(self.get_options().exclude_coordinates) + ")"
```

Comment:
I think it'd make sense to memoize the compiled regex rather than the string, ie, `re.compile(...)`.

Also, instead of combining the patterns together into a single regex, you could compile each pattern separately and then check in `_is_update_coordinate` with each. Then the debug statement could point to the exact option value that caused the exclusion.